File: Client/src/main/java/edu/najah/server/AdminServer.java
package edu.najah.server;

import java.io.FileOutputStream;
import java.nio.file.Path;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.util.Base64;
import java.util.UUID;

import static java.nio.file.Files.delete;

public class AdminServer {
  public static String login(String email, String password) {
    try {
      String result = "";
      Class.forName("com.mysql.cj.jdbc.Driver");
      Connection connection = DriverManager.getConnection("jdbc:mysql://localhost/easyproject", "root", "");
      PreparedStatement preparedStatement = connection.prepareStatement("SELECT * FROM users WHERE email='" + email + "' AND password='" + password + "' AND isAdmin='1'");
      ResultSet resultSet = preparedStatement.executeQuery();
      if (resultSet.next()) {
        result += "id:" + resultSet.getString("id") + ";";
        result += "email:" + resultSet.getString("email") + ";";
        result += "password:" + resultSet.getString("password") + ";";
        result += "name:" + resultSet.getString("name") + ";";
        result += "address:" + resultSet.getString("address") + ";";
        result += "photo:C:/xampp/htdocs/Server" + resultSet.getString("photo").substring(1) + ";";
        result += "isAdmin:" + resultSet.getString("isAdmin") + ";";
        return result;
      }
      return "Email or Password is incorrect";
    } catch (Exception e) {
      e.printStackTrace();
      return "Login Failed";
    }
  }
  
  public static String updateInfo(String id, String email, String password, String name, String address) {
    try {
      Class.forName("com.mysql.cj.jdbc.Driver");
      Connection connection = DriverManager.getConnection("jdbc:mysql://localhost/easyproject", "root", "");
      PreparedStatement preparedStatement = connection.prepareStatement("Select email from users where id='" + id + "'");
      ResultSet resultSet = preparedStatement.executeQuery();
      
      // check if email is not changed
      if (resultSet.next() && resultSet.getString("email").equals(email)) {
        preparedStatement = connection.prepareStatement("UPDATE users SET password='" + password + "', name='" + name + "', address='" + address + "' WHERE id='" + id + "'");
        return preparedStatement.executeUpdate() > 0 ? "Updated Successfully" : "Updated Failed";
      }
      
      // check if email is changed
      preparedStatement = connection.prepareStatement("Select email from users where email='" + email + "'");
      resultSet = preparedStatement.executeQuery();
      if (resultSet.next()) {
        return "Email is already taken!";
      }
      // if email is not exist
      preparedStatement = connection.prepareStatement("UPDATE users SET email='" + email + "', password='" + password + "', name='" + name + "', address='" + address + "' WHERE id='" + id + "'");
      return preparedStatement.executeUpdate() > 0 ? "Updated Successfully" : "Updated Failed";
    } catch (Exception e) {
      e.printStackTrace();
      return "Updated Failed";
    }
  }
  
  public static String updateImage(String photoExtension, String id, String photo) {
    Connection connection;
    PreparedStatement preparedStatement;
    ResultSet resultSet;
    try {
      String imagePath = "./Images/" + UUID.randomUUID() + "." + photoExtension;
      Class.forName("com.mysql.cj.jdbc.Driver");
      connection = DriverManager.getConnection("jdbc:mysql://localhost/easyproject", "root", "");
      preparedStatement = connection.prepareStatement("Select photo from users where id='" + id + "'");
      resultSet = preparedStatement.executeQuery();
      if (resultSet.next())
        delete(Path.of("C:/xampp/htdocs/Server" + resultSet.getString("photo").substring(1))); // unlink the old photo
      byte[] decodedImage = Base64.getDecoder().decode(photo);
      FileOutputStream fileOutputStream = new FileOutputStream("C:/xampp/htdocs/Server" + imagePath.substring(1));
      fileOutputStream.write(decodedImage);
      fileOutputStream.close();
      preparedStatement = connection.prepareStatement("UPDATE users SET photo='" + imagePath + "' WHERE id='" + id + "'");
      return preparedStatement.executeUpdate() > 0 ? "Updated Successfully" : "Updated Failed";
    } catch (Exception e) {
      e.printStackTrace();
      return "Login Failed";
    }
  }
  
  public static String addNewUser(String email, String password, String name, String address, String id, String photo, String photoExtension, String isAdmin) {
    Connection connection;
    PreparedStatement preparedStatement;
    ResultSet resultSet;
    try {
      String imagePath = "./Images/" + UUID.randomUUID() + "." + photoExtension;
      Class.forName("com.mysql.cj.jdbc.Driver");
      connection = DriverManager.getConnection("jdbc:mysql://localhost/easyproject", "root", "");
      
      // if id is given, check if the id is already exist
      if (!id.equals("default")) {
        preparedStatement = connection.prepareStatement("Select id from users where id='" + id + "'");
        resultSet = preparedStatement.executeQuery();
        if (resultSet.next()) return "ID is already taken!";
      }
      
      // check if email is already exist
      preparedStatement = connection.prepareStatement("Select email from users where email='" + email + "'");
      resultSet = preparedStatement.executeQuery();
      if (resultSet.next()) return "Email is already taken!";
      
      byte[] decodedImage = Base64.getDecoder().decode(photo);
      FileOutputStream fileOutputStream = new FileOutputStream("C:/xampp/htdocs/Server" + imagePath.substring(1));
      fileOutputStream.write(decodedImage);
      fileOutputStream.close();
      
      // insert new user
      if (!id.equals("default"))
        preparedStatement = connection.prepareStatement("INSERT INTO users (email, password, name, address, photo, isAdmin, id) VALUES (?, ?, ?, ?, ?, ?, ?)");
      else
        preparedStatement = connection.prepareStatement("INSERT INTO users (email, password, name, address, photo, isAdmin) VALUES (?, ?, ?, ?, ?, ?)");
      preparedStatement.setString(1, email);
      preparedStatement.setString(2, password);
      preparedStatement.setString(3, name);
      preparedStatement.setString(4, address);
      preparedStatement.setString(5, imagePath);
      preparedStatement.setString(6, isAdmin);
      if (!id.equals("default")) preparedStatement.setString(7, id);
      return preparedStatement.executeUpdate() > 0 ? "Added Successfully" : "Added Failed";
    } catch (Exception e) {
      e.printStackTrace();
      return "Added Failed";
    }
  }
  
  public static String getUsersFromDB(String usersType, String withSearch, String searchBy, String searchField) {
    Connection connection;
    PreparedStatement preparedStatement;
    ResultSet resultSet;
    try {
      Class.forName("com.mysql.cj.jdbc.Driver");
      connection = DriverManager.getConnection("jdbc:mysql://localhost/easyproject", "root", "");
      
      if (withSearch.equals("1") && !searchBy.equals("none")) {
        if (searchBy.equals("id")) {
          if (usersType.equals("all"))
            preparedStatement = connection.prepareStatement("Select * from users where id like '%" + searchField + "%'");
          else
            preparedStatement = connection.prepareStatement("Select * from users where id like '%" + searchField + "%' and isAdmin='" + usersType + "'");
        } else {
          if (usersType.equals("all"))
            preparedStatement = connection.prepareStatement("Select * from users where name like '%" + searchField + "%'");
          else
            preparedStatement = connection.prepareStatement("Select * from users where name like '%" + searchField + "%' and isAdmin='" + usersType + "'");
        }
      } else {
        if (usersType.equals("all")) preparedStatement = connection.prepareStatement("Select * from users");
        else preparedStatement = connection.prepareStatement("Select * from users where isAdmin='" + usersType + "'");
      }
      
      resultSet = preparedStatement.executeQuery();
      StringBuilder users = new StringBuilder();
      while (resultSet.next())
        users.append(resultSet.getString("id")).append(",").append(resultSet.getString("email")).append(",").append(resultSet.getString("password")).append(",").append(resultSet.getString("name")).append(",").append(resultSet.getString("address")).append(",").append("C:/xampp/htdocs/Server").append(resultSet.getString("photo").substring(1)).append(",").append(resultSet.getString("isAdmin")).append(";");
      return users.toString();
    } catch (Exception e) {
      e.printStackTrace();
      return "Query Failed";
    }
  }
  
  public static String deleteUser(String id) {
    Connection connection;
    PreparedStatement preparedStatement;
    ResultSet resultSet;
    try {
      Class.forName("com.mysql.cj.jdbc.Driver");
      connection = DriverManager.getConnection("jdbc:mysql://localhost/easyproject", "root", "");
      preparedStatement = connection.prepareStatement("Delete from users where id='" + id + "'");
      return preparedStatement.executeUpdate() > 0 ? "Deleted Successfully" : "Deleted Failed";
    } catch (Exception e) {
      e.printStackTrace();
      return "Query Failed";
    }
  }
}

File: Server/AdminServer.php
<?php
function login($email, $password) {
  $conn = new mysqli("localhost", "root", "", "easyproject");
  $loginQuery = "SELECT * FROM users WHERE email='" . $email . "' AND password='" . $password . "' AND isAdmin='1'";
  $result = mysqli_query($conn, $loginQuery);
  if (mysqli_num_rows($result) > 0) {
    $str = "";
    foreach (mysqli_fetch_assoc($result) as $key => $value) {
      if ($key == "photo") {
        $path = new SplFileInfo($value);
        $str .= $key . ":" . $path->getRealPath() . ",";
      } else {
        $str .= $key . ":" . $value . ";";
      }
    }
    return $str;
  } else {
    return "Email or Password is incorrect";
  }
}

function updateInfo($id, $email, $password, $name, $address) {
  $conn = new mysqli("localhost", "root", "", "easyproject");
  $updateQuery = "Select email from users where id='" . $id . "'";
  $result = mysqli_query($conn, $updateQuery);
  $row = mysqli_fetch_row($result)[0];
  
  if ($row == $email) { // if email is not changed
    $updateQuery = "UPDATE users SET password='" . $password . "', name='" . $name . "', address='" . $address . "' WHERE id='" . $id . "'";
    return (mysqli_query($conn, $updateQuery) ? "Updated Successfully" : "Updated Failed");
  }
  // check if email is changed
  $updateQuery = "SELECT * FROM users WHERE email='" . $email . "'";
  $result = mysqli_query($conn, $updateQuery);
  if (mysqli_num_rows($result) > 0) {
    return "Email is already taken!";
  }
  $updateQuery = "UPDATE users SET email='" . $email . "', password='" . $password . "', name='" . $name . "', address='" . $address . "' WHERE id='" . $id . "'";
  return (mysqli_query($conn, $updateQuery) ? "Updated Successfully" : "Updated Failed");
}

function updateImage($imagePath, $id, $photo) {
  $conn = new mysqli("localhost", "root", "", "easyproject");
  $updateQuery = "Select photo from users where id='" . $id . "'";
  $result = mysqli_query($conn, $updateQuery);
  $row = mysqli_fetch_row($result)[0];
  unlink($row);
  file_put_contents($imagePath, base64_decode($photo));
  $updateQuery = "UPDATE users SET photo='$imagePath' WHERE id='$id'";
  return (mysqli_query($conn, $updateQuery) ? "Updated Successfully" : "Updated Failed");
}

function addNewUser($email, $password, $name, $address, $id, $photo, $isAdmin, $imagePath) {
  $conn = new mysqli("localhost", "root", "", "easyproject");
  
  if ($id != "default") { // if id is given, check if it is already exist
    $ids = array();
    $getIdsQuery = "SELECT id FROM users";
    $result = mysqli_query($conn, $getIdsQuery);
    while ($row1 = mysqli_fetch_assoc($result)) {
      $ids[] = $row1['id'];
    }
    if (in_array($id, $ids)) { // if id is already exist
      return "ID is already taken!";
    }
  }
  
  // check if email is already exist
  $emails = array();
  $getEmailsQuery = "SELECT email FROM users";
  $result = mysqli_query($conn, $getEmailsQuery);
  while ($row1 = mysqli_fetch_assoc($result)) {
    $emails[] = $row1['email'];
  }
  if (in_array($email, $emails)) { // if email is already exist
    return "This email is already taken!";
  }
  
  file_put_contents($imagePath, base64_decode($photo));
  if ($id != "default")
    $insertQuery = "INSERT INTO users (email, password, name, address, id, photo, isAdmin) VALUES ('$email', '$password', '$name', '$address', '$id', '$imagePath', '$isAdmin')";
  else
    $insertQuery = "INSERT INTO users (email, password, name, address, photo, isAdmin) VALUES ('$email', '$password', '$name', '$address', '$imagePath', '$isAdmin')";
  return (mysqli_query($conn, $insertQuery) ? "Added Successfully" : "Added Failed");
}

function getUsersFromDB($usersType, $withSearch, $searchBy, $searchField) {
  $conn = new mysqli("localhost", "root", "", "easyproject");
  if ($withSearch == "1" && $searchBy != "none") {
    if ($searchBy == "id") {
      if ($usersType == "all") {
        $getUsersQuery = "SELECT * FROM users WHERE id like '%$searchField%'";
      } else {
        $getUsersQuery = "SELECT * FROM users WHERE id like '%$searchField%' AND isAdmin='$usersType'";
      }
    } else {
      if ($usersType == "all")
        $getUsersQuery = "SELECT * FROM users WHERE name like '%$searchField%'";
      else
        $getUsersQuery = "SELECT * FROM users WHERE name like '%$searchField%' AND isAdmin='$usersType'";
    }
  } else {
    if ($usersType == "all")
      $getUsersQuery = "SELECT * FROM users";
    else
      $getUsersQuery = "SELECT * FROM users WHERE isAdmin='$usersType'";
  }
  
  $result = mysqli_query($conn, $getUsersQuery);
  $users = array();
  while ($row = mysqli_fetch_assoc($result)) $users[] = $row;
  $result = '';
  foreach ($users as $user) {
    $user["photo"] = realpath($user["photo"]);
    $result .= implode(',', $user) . ";";
  }
  return $result;
}

function deleteUser($id) {
  $conn = new mysqli("localhost", "root", "", "easyproject");
  $deleteQuery = "DELETE FROM users WHERE id='$id'";
  return (mysqli_query($conn, $deleteQuery) ? "Deleted Successfully" : "Deleted Failed");
}

switch ($_REQUEST['function']) {
  case 'login':
    echo login($_REQUEST['email'], $_REQUEST['password']);
    break;
  case 'updateInfo':
    echo updateInfo($_REQUEST['id'], $_REQUEST['email'], $_REQUEST['password'], $_REQUEST['name'], $_REQUEST['address']);
    break;
  case 'updateImage':
    $imagePath = './Images/' . uniqid(rand(), true) . '.' . $_REQUEST['photoExtension'];
    echo updateImage($imagePath, $_REQUEST['id'], $_REQUEST['photo']);
    break;
  case 'addNewUser':
    $imagePath = './Images/' . uniqid(rand(), true) . '.' . $_REQUEST['photoExtension'];
    echo addNewUser($_REQUEST['email'], $_REQUEST['password'], $_REQUEST['name'], $_REQUEST['address'], $_REQUEST['id'], $_REQUEST['photo'], $_REQUEST['isAdmin'], $imagePath);
    break;
  case 'getUsersFromDB':
    echo getUsersFromDB($_REQUEST['usersType'], $_REQUEST['withSearch'], $_REQUEST['searchBy'], $_REQUEST['searchField']);
    break;
  case 'deleteUser':
    echo deleteUser($_REQUEST['id']);
    break;
  default:
    echo "Invalid Request";
}